lib/npg_tracking/glossary/moniker.pm

```diff
@@ -35,6 +38,45 @@ sub file_name {
   return $name;
 }
 
+sub file_name_full {
+  my ($self, $name, @options) = @_;
+
+  $name or croak 'File name base should be given';
+
+  my %h;
+  if (@options) {
+    try {
+      use warnings FATAL => qw/misc/; # Turn warning
+                                      # 'Odd number of elements in hash assignment'
+                                      # into an error...
+      %h = @options;
+    } catch {
+      my $e = $_;
+      if ($e =~ /Odd\ number\ of\ elements\ in\ hash\ assignment/smxi) {
+        $e = 'The argument list should contain a file name'.
+             ' and, optionally, a list of key-value options';
+      }
+      croak $e;
+    };
```

Comment:
Why a try/catch, and not simply check that the length of @options is divisible by 2, before assigning it to %h?